tools/github-bot/README.md

```diff
@@ -1,33 +1,81 @@
+> A GitHub App built with [Probot](https://github.com/probot/probot) to enhance developer experience on the ledger-live repository
 
-## Setup
+In short, `live-github-bot` is a tool deployed to Vercel which accepts incoming requests based on some Github Events.  
+It provides us with useful actions and tools we can provide for developers, as well as a deeper developer experience when working on Ledger Live mono repo.  
 
-```sh
-# Install dependencies
-npm install
 
-# Run the bot
-npm start
+## Code Architecture
+
+We've structured the code to make it work as **plugins** that extend the github app.
+
+```
+src
+├── commands : new functionality added to github through commenting or CTAs in check runs reports
+├── features : features that changes the behavior and enhance PR checks and reports (eg: linting commit messages, auto close PR...)
+├── tools    : common utilities
+└── index.ts : entry point
 ```
 
-## Docker
+The `index.ts` is where we activate the wanted plugins, keeping it simple and clean.
+All logic related stuff either goes into `features` or `commands` folders, depending on what we want to implement.
+
+## Continuous Integration Orchestration
+
+Being restricted by Github's default behavior in the context of a mono repo, we could not have a clean output for our CI jobs (doppleganger jobs for required checks, impossibility to have more granular CI...).  
+Thus, instead on relying on Github default event to create report on the pull requests, we've decided to tailor the experience to our needs.
+
+> Usually, workflows (CI jobs) are triggered on `pull_request` or `push` events
+
+In the case of a mono repository, with linked dependencies between the different packages, it was not possible to tell Github CI to only accept jobs on the **affected** code.
+Add to that the missing required checks if you skip some specific jobs because the code was not affected, and we are left with a PR in limbo state, with pending required checks that will never happen.
+
+To circumvent that, we use the bot to orchestrate our CI.
+
+### Github Runs and Checks
 
-```sh
-# 1. Build container
-docker build -t live-github-bot .
+When using the _basic_ behavior of the `push` and `pull_request` event, **checks** are created for each job in all the triggered workflow, and are shown on the PR with their current status (pending, success, error...).
+It bloaters the check list with a lot of unneeded notifications, and doesn't provide much informations.
+**Checks** can actually create by the github bot, and enhance the outputs of our check.
 
-# 2. Start container
-docker run -e APP_ID=<app-id> -e PRIVATE_KEY=<pem-value> live-github-bot
+To prevent workflows from creating checks automatically, the trigger of the CI is not on `push` or `pull_request` anymore but on `workflow_dispatch`.
+Giving the `workflow_dispatch` as trigger for a workflow, when can then trigger it manually. This means the workflow will run decoralated from any PR, and thus won't create checks.
```

Comment:
I'm not sure I understand this one also:

"Giving the `workflow_dispatch` as trigger for a workflow, when can then trigger it manually."

And:

decoralated -> decorrelated